TST: pyinstaller cron job failed with MATPLOTLIBDATA environment variable was deprecated

Example log: https://travis-ci.org/github/astropy/astropy/jobs/731532639

```
ImportError while loading conftest '/home/travis/build/astropy/astropy/.pyinstaller/astropy_tests/conftest.py'.
_pytest/config/__init__.py:570: in _importconftest
    ???
_pytest/pathlib.py:517: in import_path
    ???
_pytest/assertion/rewrite.py:171: in exec_module
    ???
astropy_tests/conftest.py:22: in <module>
    import matplotlib
../.tox/pyinstaller/lib/python3.8/site-packages/PyInstaller/loader/pyimod03_importers.py:489: in exec_module
    exec(bytecode, module.__dict__)
matplotlib/__init__.py:1015: in <module>
    ???
matplotlib/__init__.py:878: in rc_params
    ???
matplotlib/__init__.py:731: in matplotlib_fname
    ???
matplotlib/__init__.py:729: in gen_candidates
    ???
matplotlib/__init__.py:276: in wrapper
    ???
matplotlib/__init__.py:653: in _get_data_path
    ???
matplotlib/cbook/deprecation.py:111: in warn_deprecated
    ???
matplotlib/cbook/__init__.py:1993: in _warn_external
    ???
E   matplotlib.cbook.deprecation.MatplotlibDeprecationWarning:
E   The MATPLOTLIBDATA environment variable was deprecated in Matplotlib 3.1 and will be removed in 3.3.
ERROR: InvocationError for command /home/travis/build/astropy/astropy/.pyinstaller/run_astropy_tests (exited with code 4)
```

Naive re-pinning didn't work (#10779).

Also see: #10639

--- a/.gitignore
+++ b/.gitignore
@@ -43,6 +43,12 @@ distribute-*.tar.gz
 .venv
 venv
 
+# pyinstaller files
+.pyinstaller/astropy_tests/
+.pyinstaller/run_astropy_tests
+.pyinstaller/run_astropy_tests.spec
+
+
 # Other
 .cache
 .tox

--- a/astropy/conftest.py
+++ b/astropy/conftest.py
@@ -6,6 +6,7 @@ making use of astropy's test runner).
 """
 import os
 import builtins
+import sys
 import tempfile
 import warnings
 
@@ -19,6 +20,15 @@ import pytest
 from astropy.tests.helper import enable_deprecations_as_exceptions
 
 try:
+    # This is needed to silence a warning from matplotlib caused by
+    # PyInstaller's matplotlib runtime hook.  This can be removed once the
+    # issue is fixed upstream in PyInstaller, and only impacts us when running
+    # the tests from a PyInstaller bundle.
+    # See https://github.com/astropy/astropy/issues/10785
+    if getattr(sys, 'frozen', False) and hasattr(sys, '_MEIPASS'):
+        # The above checks whether we are running in a PyInstaller bundle.
+        warnings.filterwarnings("ignore", "(?s).*MATPLOTLIBDATA.*",
+                                category=UserWarning)
     import matplotlib
 except ImportError:
     HAS_MATPLOTLIB = False

